pkg/backends/newrelic/newrelic.go

```diff
@@ -307,13 +327,31 @@ func (n *Client) post(ctx context.Context, buffer *bytes.Buffer, data interface{
 	b.Clock = clck
 	b.Reset()
 	b.MaxElapsedTime = n.maxRequestElapsedTime
+
+	var next time.Duration
+	var retryAfterErr *RetryAfterError
+	canRetryRateLimit := true
 	for {
 		if err = post(); err == nil {
 			atomic.AddUint64(&n.batchesSent, 1)
 			return nil
 		}
 
-		next := b.NextBackOff()
+		// If New Relic limitted us we should retry after
```

Comment:
This will retry only one (if we're in the 429 case), which may be intended, but it also means we're ignoring the configured `n.maxRequestElapsedTime`.

There's two concerns:
- Should we keep retrying until we reach the configured maximum, or only retry once?
- We should never retry for longer than the configured `maxRequestElapsedTime`, even if NR tells us to wait that long.

I would suggest using a pattern more like:
```
next := b.NextBackOff()
if next != backoff.Stop && errors.As(err, &retryAfterError) {
  next = max(next, retryAfterError.RetryAfter())
  if n.maxRequestElapsedTime > 0 {
    // this is not ideal, because it means we could potentially sleep for 2*n.maxRequestElapsedTime,
    // but it does put an upper bound on the duration.  The proper solution would be to track the
    // start time of the overall request, and make sure we don't sleep past that, but this is good enough.
    next = min(next, n.maxRequestElapsedTime)
  }
}
```

Reply:
Thanks for your review.

Indeed, I was not sure about putting a limit on the number of retries. I think, the approach you mentioned covers more and use `maxRequestElapsedTime`, I will use that.

Reply:
Implemented in last commit